firmware/quadruna/VC/src/app/vehicle_dynamics/app_torqueVectoring.c

```diff
@@ -170,3 +121,163 @@ void app_torqueVectoring_handleAcceleration(void)
+
+    // initializing active diff outputs
+    traction_control_outputs.torque_left_Nm  = 0.0;
+    traction_control_outputs.torque_right_Nm = 0.0;
+
+    if (accelerator_pedal_percent <= PEDAL_NOT_PRESSED && torqueVectoringInternalState != OFF)
+    {
+        torqueVectoringInternalState = noPedalInput();
+        return;
+    }
+
+    switch (torqueVectoringInternalState)
+    {
+        case OFF:
+
+            if (accelerator_pedal_percent > 0.0f)
+            {
+                if (TRACTION_CONTROL_LOWER_BOUND <= steering_angle_deg &&
+                    steering_angle_deg <= TRACTION_CONTROL_UPPER_BOUND)
+                {
+                    torqueVectoringInternalState = runTractionControl(power_limit);
+                }
+                else
+                {
+                    torqueVectoringInternalState = runActiveDiff(power_limit);
+                }
+            }
+
+            break;
+
+        case TRACTION_CONTROL:
+
+            if (!run_traction_control)
+            {
+                torqueVectoringInternalState = runActiveDiff(power_limit);
+                break;
+            }
+
+            // we do not nessecarily want hystersis to enter active diff from traction control as if we increase the
+            // window for to be activate for a larger angle window we would be further assuming that there are little
+            // lateral forces in play at a steering angle where they very much may be in play
+            if (TRACTION_CONTROL_LOWER_BOUND <= steering_angle_deg &&
+                steering_angle_deg <= TRACTION_CONTROL_UPPER_BOUND)
+            {
+                torqueVectoringInternalState = runTractionControl(power_limit);
+            }
+
+            else
+            {
+                torqueVectoringInternalState = runActiveDiff(power_limit);
+            }
+            break;
+
+        case ACTIVE_DIFF:
+
+            if (run_traction_control)
+            {
+                // we implement hystersis here as we do not want to oscillate back into slip control by accident
+                if ((TRACTION_CONTROL_LOWER_BOUND + STEERING_ANG_HYSTERSIS) <= steering_angle_deg &&
+                    steering_angle_deg <= (TRACTION_CONTROL_UPPER_BOUND - STEERING_ANG_HYSTERSIS))
+                {
+                    torqueVectoringInternalState = runTractionControl(power_limit);
+                }
+            }
+
+            else
+            {
+                torqueVectoringInternalState = runActiveDiff(power_limit);
+            }
+            break;
+
+        default:
+            torqueVectoringInternalState = noPedalInput();
+            break;
+    }
+
+    return;
+}
+
+/* Event Handlers */
+
+static TVInteralStates noPedalInput()
+{
+    app_canTx_VC_LeftInverterTorqueCommand_set(0.0f);
+    app_canTx_VC_RightInverterTorqueCommand_set(0.0f);
+    return OFF;
+}
+
+static TVInteralStates runTractionControl(float power_limit)
+{
+    if (torqueVectoringInternalState != TRACTION_CONTROL)
```

Comment:
i mean like ok?? so your state isn't traction control but you are running traction control code because it's the first time?

Reply:
i think that's a lot of mental gynamistics just to not have a run on entry function